NPE when postgres db disappears

At some point local network had a hiccup and the postgresql db disappeared.
clair daemon died with a nil pointer exception.

```
2016-08-13 06:23:48.772735 E | pgsql: updateLock: dial tcp: lookup stinky.usersys on 10.11.5.19:53: no such host
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x6677b6]

goroutine 808 [running]:
panic(0x982a20, 0xc420018110)
        /home/vbatts/src/go/src/runtime/panic.go:500 +0x1a1
database/sql.(*Tx).Rollback(0x0, 0x0, 0xe7dae0)
        /home/vbatts/src/go/src/database/sql/sql.go:1264 +0x26
github.com/coreos/clair/database/pgsql.(*pgSQL).insertVulnerability(0xc4200105c0, 0x0, 0xc423968fc0, 0xd, 0x0, 0xc4284a4790, 0xf, 0xc421735680, 0x11b, 0xc425d3ea80, ...)
        /home/vbatts/src/github.com/coreos/clair/database/pgsql/vulnerability.go:224 +0x1362
github.com/coreos/clair/database/pgsql.(*pgSQL).InsertVulnerabilities(0xc4200105c0, 0xc42cbc6000, 0x166c2, 0x1a880, 0x0, 0x1, 0xc42591a1b0)
        /home/vbatts/src/github.com/coreos/clair/database/pgsql/vulnerability.go:183 +0xea
github.com/coreos/clair/updater.Update(0xe87ca0, 0xc4200105c0, 0xc4202ec901)
        /home/vbatts/src/github.com/coreos/clair/updater/updater.go:172 +0x280
github.com/coreos/clair/updater.Run.func1(0xe87ca0, 0xc4200105c0, 0x1, 0xc420b2a0e0)
        /home/vbatts/src/github.com/coreos/clair/updater/updater.go:103 +0x3e
created by github.com/coreos/clair/updater.Run
        /home/vbatts/src/github.com/coreos/clair/updater/updater.go:105 +0x840
exit status 2
```

Versions:
postgresql-server-9.5.3-1.fc24.x86_64
clair commit 4bc6416132dd5d3586d08a17d358adeeb760269e

--- a/database/pgsql/feature.go
+++ b/database/pgsql/feature.go
@@ -112,7 +112,9 @@ func (pgSQL *pgSQL) insertFeatureVersion(fv database.FeatureVersion) (id int, er
 	// Begin transaction.
 	tx, err := pgSQL.Begin()
 	if err != nil {
-		tx.Rollback()
+		if tx != nil {
+			tx.Rollback()
+		}
 		return 0, handleError("insertFeatureVersion.Begin()", err)
 	}
 

--- a/database/pgsql/layer.go
+++ b/database/pgsql/layer.go
@@ -87,6 +87,9 @@ func (pgSQL *pgSQL) FindLayer(name string, withFeatures, withVulnerabilities boo
 		// preferred to use a nested loop.
 		tx, err := pgSQL.Begin()
 		if err != nil {
+			if tx != nil {
+				tx.Rollback()
+			}
 			return layer, handleError("FindLayer.Begin()", err)
 		}
 		defer tx.Commit()
@@ -295,7 +298,9 @@ func (pgSQL *pgSQL) InsertLayer(layer database.Layer) error {
 	// Begin transaction.
 	tx, err := pgSQL.Begin()
 	if err != nil {
-		tx.Rollback()
+		if tx != nil {
+			tx.Rollback()
+		}
 		return handleError("InsertLayer.Begin()", err)
 	}
 

--- a/database/pgsql/notification.go
+++ b/database/pgsql/notification.go
@@ -173,6 +173,9 @@ func (pgSQL *pgSQL) loadLayerIntroducingVulnerability(vulnerability *database.Vu
 	// allowing a small nested loop join instead.
 	tx, err := pgSQL.Begin()
 	if err != nil {
+		if tx != nil {
+			tx.Rollback()
+		}
 		return -1, handleError("searchNotificationLayerIntroducingVulnerability.Begin()", err)
 	}
 	defer tx.Commit()

--- a/database/pgsql/vulnerability.go
+++ b/database/pgsql/vulnerability.go
@@ -257,7 +257,9 @@ func (pgSQL *pgSQL) insertVulnerability(vulnerability database.Vulnerability, on
 	// Begin transaction.
 	tx, err := pgSQL.Begin()
 	if err != nil {
-		tx.Rollback()
+		if tx != nil {
+			tx.Rollback()
+		}
 		return handleError("insertVulnerability.Begin()", err)
 	}
 
@@ -588,7 +590,9 @@ func (pgSQL *pgSQL) DeleteVulnerability(namespaceName, name string) error {
 	// Begin transaction.
 	tx, err := pgSQL.Begin()
 	if err != nil {
-		tx.Rollback()
+		if tx != nil {
+			tx.Rollback()
+		}
 		return handleError("DeleteVulnerability.Begin()", err)
 	}
 

